MoreLinq/GroupAdjacent.cs

```diff
@@ -253,7 +253,7 @@ static partial class MoreEnumerable
 
             // This should be removed once the target framework is bumped to something that supports covariance
             TResult ResultSelectorWrapper(TKey key, IList<TSource> group) => resultSelector(key, group);
-            return GroupAdjacentImpl(source, keySelector, i => i, ResultSelectorWrapper,
+            return GroupAdjacentImpl(source, keySelector, Identity<TSource>, ResultSelectorWrapper,
```

Comment:
Type argument is redundant:

```suggestion
            return GroupAdjacentImpl(source, keySelector, Identity, ResultSelectorWrapper,
```